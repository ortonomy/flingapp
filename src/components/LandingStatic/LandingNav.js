// basic react imports
import React, { Component }from 'react';

// library imports
import { connect } from 'react-redux';

// component dependencies
import LandingNavItems from './LandingNavItems';

// styles
import styles from './LandingNav.module.css';

// assets
import WebTitle from '../../assets/images/web-title@svg.svg';

@connect(
  state => (
    {
      Login: state.Login
    }
  ),
  null
)
class LandingNav extends Component {
  constructor(props) {
    super(props);

    this.navItems = this.props.Login.loggedIn === true ? [
      {
        href: '/main',
        text: `Go to app`
      },
      {
        href: '/logout',
        text: 'Log out'
      }
    ] : [
      {
        href: '/login',
        text: `Log in`
      },
      {
        href: '/register',
        text: 'Register'
      }
    ];
  }
  render() {
    return (
      <div className={styles.NavBar}>
        <img className={styles.NavBarLogo} src={WebTitle} alt='small logo' />
        <LandingNavItems items={this.navItems} />
      </div>
    )
  }
}

export default LandingNav;