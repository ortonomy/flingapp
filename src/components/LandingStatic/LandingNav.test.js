import React from 'react';
import LandingNav from './LandingNav';

it('renders without crashing', () => {
  const wrapper = shallow(<LandingNav />);
  expect(wrapper).toMatchSnapshot();
});
