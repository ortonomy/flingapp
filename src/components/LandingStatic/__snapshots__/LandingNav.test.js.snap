// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingNav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="small logo"
          className={undefined}
          src="web-title@svg.svg"
/>,
        <LandingNavItems
          items={
                    Array [
                              Object {
                                "href": "/login",
                                "text": "Log in",
                              },
                              Object {
                                "href": "/register",
                                "text": "Register",
                              },
                            ]
          }
/>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "small logo",
          "className": undefined,
          "src": "web-title@svg.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "items": Array [
            Object {
              "href": "/login",
              "text": "Log in",
            },
            Object {
              "href": "/register",
              "text": "Register",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="small logo"
            className={undefined}
            src="web-title@svg.svg"
/>,
          <LandingNavItems
            items={
                        Array [
                                    Object {
                                      "href": "/login",
                                      "text": "Log in",
                                    },
                                    Object {
                                      "href": "/register",
                                      "text": "Register",
                                    },
                                  ]
            }
/>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "small logo",
            "className": undefined,
            "src": "web-title@svg.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "items": Array [
              Object {
                "href": "/login",
                "text": "Log in",
              },
              Object {
                "href": "/register",
                "text": "Register",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
